### PR TITLE
Fixed keywords disapear from search index

### PR DESCRIPTION
**Pull request by Casper-O**
http://forge.prestashop.com/browse/PSCSX-8239#

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x (But should be on 1.7 also)
| Description?  | Delete a product, and you delete words that is shared in search_word tabel, make other product disapear from search until next full reindex.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | No
| Deprecations? | No
| Fixed ticket? | http://forge.prestashop.com/browse/PSCSX-8239#
| How to test?  | see my comment in ticket, link with db Fiddle etc

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8587)

### CODE REVIEW

**Comment by Casper-O** [comment]
Just to be clear, what I did was to revert back to how it was back in 2015. I can't get up with a smarter or better solution.

**Comment by Casper-O** [comment]
shocked to learn this did not make it for 1.6.1.18 - also can't see whats wrong with the code checks?

**Comment by Casper-O** [comment]
@mickaelandrieu what needs to be fixed to make the review possible? i did like to see it in 1.6.1.19 - but its not added to the milestone...

File: classes/Product.php [hunk 1]
```diff
@@ -1890,10 +1890,16 @@ public function deleteSearchIndexes()
...
+                'DELETE FROM `'._DB_PREFIX_.'search_index`
+				WHERE `id_product` = '.(int)$this->id
+            )
+            &&
+            Db::getInstance()->execute(
+                'DELETE FROM `'._DB_PREFIX_.'search_word`
+				WHERE `id_word` NOT IN (
+					SELECT id_word
+					FROM `'._DB_PREFIX_.'search_index`
+				)'
```
Comment (jocel1) [inline classes/Product.php]: Please use

```DELETE sw FROM `'._DB_PREFIX_.'search_word` sw LEFT JOIN `'._DB_PREFIX_.'search_index` si ON (sw.id_word=si.id_word) WHERE si.id_word IS NULL; ```

which should be much better for performances

**Review by jocel1 — CHANGES_REQUESTED** [review]

**Comment by Casper-O** [comment]
Hi Thanks for your feedback, i Will test the changes later today and Update my PR after

**Comment by mickaelandrieu** [comment]
@Casper-O it's not up to me to add something in a release... wdyt @eternoendless ? Can we plan it for 1.6.1.19?

**Comment by Casper-O** [comment]
@jocel1 I tested the changes you asked for and it worked as expected. I did not check performance but I think you are right.

@mickaelandrieu sorry that i ping you regarding this, you where just adding tags etc to this PR so i thought you could push it on. It seems for me like not just a minor bug (that is how it is tagged at Forge), that a store search function is "useless" if you delete a product, until next time you make a full reindex of your products. That is why i'm pushing for this changes.

I can also remember i checked 1.7.X 3 months ago and the same code is there. I did not check how most recent version of 1.7 handle this, and if issue is resolved there yet.

**Comment by Casper-O** [comment]
@jocel1 could you confirm the changes have been made as requested - then it could hopefully make it for 1.6.1.19

**Comment by Casper-O** [comment]
Another version, another 3 months passed for an issue that is actually critical since it destroys your visitors option to search your products...

**Comment by Casper-O** [comment]
@jocel1 the requested changes have been made... Would you please review it and accept it...

If nothing happens soon i will just close the PR and the issue. I really dont feel like wasting another year waiting for a fix to be accepted

**Review by jocel1 — APPROVED** [review]

**Review by jocel1 — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8587)* status: all files reviewed, 1 unresolved discussion (waiting on @jocel1)

**Review by jocel1 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8587)* status: :shipit: complete! all files reviewed, all discussions resolved

**Comment by jocel1** [comment]
Hi @Casper-O,
Sorry for the delay, my mail filters were crappy and I missed your fix :/ Approved!

**Comment by Casper-O** [comment]
closed as https://github.com/PrestaShop/PrestaShop/issues/10530 - im done trying help fix issues

**Pull request closed without merging**